modules/accounts/email.nix

```
@@ -114,6 +114,26 @@ let
     };
   };
 
+  authenticationOption = mkOption {
+    type = types.nullOr (
+      types.enum [
+        "anonymous"
+        "apop"
+        "clear"
+        "cram_md5"
+        "digest_md5"
+        "gssapi"
+        "login"
+        "ntlm"
+        "plain"
+        "xoauth2"
+      ]
+    );
```

```suggestion
    type = types.nullOr (
      types.either types.str (
        types.enum [
          "anonymous"
          "apop"
          "clear"
          "cram_md5"
          "digest_md5"
          "gssapi"
          "login"
          "ntlm"
          "plain"
          "xoauth2"
        ]
      )
    );
```
Could make it freeform while using enum to guide a user, in case new auth types become available.